internal/controller/domain_controller.go

```diff
@@ -42,22 +60,528 @@ func (r *DomainReconciler) Reconcile(ctx context.Context, req mcreconcile.Reques
+	}
+
+	// Update condition status
+	status := metav1.ConditionFalse
+	reason := "DomainNotReady"
+	message := "Domain verification pending"
+
+	if ready {
+		status = metav1.ConditionTrue
+		reason = "DomainReady"
+		message = "Domain verified and WHOIS data fetched"
+	}
+
+	// Only update if status changed
+	if readyCondition.Status != status {
+		readyCondition.Status = status
+		readyCondition.Reason = reason
+		readyCondition.Message = message
+		readyCondition.LastTransitionTime = metav1.Now()
+	}
+}
+
+// updateDomainConditions updates all conditions for the domain
+func (r *DomainReconciler) updateDomainConditions(domain *datumapisv1alpha.Domain) {
+	// Ensure status is initialized
+	if domain.Status.Conditions == nil {
+		domain.Status.Conditions = []metav1.Condition{}
+	}
+
+	// Update verification condition
+	r.updateVerificationCondition(domain)
+
+	// Update WHOIS condition
+	r.updateWHOISCondition(domain)
+
+	// Update Ready condition (depends on verification and WHOIS)
+	r.updateReadyCondition(domain)
+}
+
+// updateVerificationCondition updates the verification condition
+func (r *DomainReconciler) updateVerificationCondition(domain *datumapisv1alpha.Domain) {
+	verified := r.isDomainVerified(domain)
+
+	// Find existing Verification condition
+	var verificationCondition *metav1.Condition
+	for i := range domain.Status.Conditions {
+		if domain.Status.Conditions[i].Type == "Verification" {
+			verificationCondition = &domain.Status.Conditions[i]
+			break
+		}
+	}
+
+	// Create new condition if it doesn't exist
+	if verificationCondition == nil {
+		verificationCondition = &metav1.Condition{
+			Type:               "Verification",
+			Status:             metav1.ConditionFalse,
+			Reason:             "VerificationPending",
+			Message:            "Waiting for DNS verification record",
+			LastTransitionTime: metav1.Now(),
+		}
+		domain.Status.Conditions = append(domain.Status.Conditions, *verificationCondition)
+	}
+
+	// Update condition status
+	status := metav1.ConditionFalse
+	reason := "VerificationPending"
+	message := "Waiting for DNS verification record"
+
+	if verified {
+		status = metav1.ConditionTrue
+		reason = "VerificationComplete"
+		message = "Domain ownership verified via DNS"
+	}
+
+	// Only update if status changed
+	if verificationCondition.Status != status {
+		verificationCondition.Status = status
+		verificationCondition.Reason = reason
+		verificationCondition.Message = message
+		verificationCondition.LastTransitionTime = metav1.Now()
+	}
+}
+
+// updateWHOISCondition updates the WHOIS condition
+func (r *DomainReconciler) updateWHOISCondition(domain *datumapisv1alpha.Domain) {
+	hasWHOISData := domain.Status.Registrar != nil
+
+	// Find existing WHOIS condition
+	var whoisCondition *metav1.Condition
+	for i := range domain.Status.Conditions {
+		if domain.Status.Conditions[i].Type == "WHOIS" {
+			whoisCondition = &domain.Status.Conditions[i]
+			break
+		}
+	}
```

Comment:
FYI you can use helper functions for working with the Status Conditions. 

- [FindStatusCondition](https://pkg.go.dev/k8s.io/apimachinery@v0.33.2/pkg/api/meta#FindStatusCondition)
- [SetStatusCondition](https://pkg.go.dev/k8s.io/apimachinery@v0.33.2/pkg/api/meta#SetStatusCondition)